### DISCUSSION TITLE
Desarrollo de código cloud-first

### DISCUSSION THREAD

**Discussion by gtheler** [GArCyAR/ReunionesAnuales#9] [Ideas]
Estimados. Les dejos algunos links sobre [FeenoX](https://www.seamplex.com/feenox/), que sería como la evolución del ya extinto [milonga](https://www.seamplex.com/milonga/).

 * https://www.seamplex.com/feenox/examples/
    - https://www.seamplex.com/feenox/examples/#iaea-2d-pwr-benchmark
    - https://www.seamplex.com/feenox/examples/#cube-spherical-bare-reactor
    - https://www.seamplex.com/feenox/examples/#illustration-of-the-xs-dilution-smearing-effect
 * https://www.seamplex.com/feenox/doc/tutorials/
    - https://www.seamplex.com/feenox/doc/tutorials/110-tensile-test/
    - https://www.seamplex.com/feenox/doc/tutorials/120-mazes/
 * https://www.seamplex.com/feenox/doc/
 * https://github.com/gtheler/2022-garcar

En particular sobre las capacidades de neutrónica, sólo tiene difusión con FEM. Milonga tenía las cuatro combinaciones difusión/SN para FEM/FVM. Cuando tenga tiempo (es open source) lo voy a implementar, pero si alguien tiene ganas de meterse, adelante. En su momento Ramiro Vignolo había hecho algo para PC y Pablo Camusso para difusión transitoria pero creo que le perdimos el rastro a estos dos. Lo mismo a otros excelentes programadores como Esteban Pellegrino, Matías Rivero, Guido Giuntoli, Juan Pablo Gómez Omil, etc. que han metido sus narices en wasora/milonga/fino en su momento. Tal vez si busco sus usernames de Github y los tageo les den ganas de volver.

Comentarios, sugerencias, pull requests, etc. son bienvenidos.

**Comment by gtheler** [comment]
En este post de LinkedIn una compañía de Finlandia muestra las capacidades de su solver:
https://www.linkedin.com/posts/alexandre-halbach_at-quanscient-we-now-solve-without-hacks-activity-7021836022753730561-zYgD?utm_source=share&utm_medium=member_desktop

Por si no fui suficientemente claro en mi presentación (me dio la impresión de que no lo fui): **la importancia de la paralelización con MPI no es que "vaya más rápido" sino poder resolver problemas de tamaños arbitrarios que no entrarían en un solo host.**

No hay mucho misterio, el solver es open source: http://sparselizard.org/
No es difícil de compilar, tal vez es un poco díficil de usar pero tampoco es nada de otro mundo.
El punto de la idea es lograr hacer eso mismo que ellos hacen para electromagnetismo pero para transporte de neutrones.

**Comment by gtheler** [comment]
El fin de semana estuve "portando" el solver SN de milonga a FeenoX.
Acá está el slab de Reed (1D): https://seamplex.com/feenox/examples/neutron_transport.html#reeds-problem

Se agradecería que compartan benchmarks 2D y 3D más copados para hacer V&V.

**Reply by marquezj** [reply]
Una compilación clásica de benchmarks es la de Barry Ganapol:

https://www.oecd-nea.org/jcms/pl_14328/analytical-benchmarks-for-nuclear-engineering-applications?details=true

**Reply by gtheler** [reply]
Que dolor de ojos. ¿No conocen LaTeX en la OECD? Tal vez es porque lo pasé muy rápido porque el formato es muy feo, pero no encontré ningún problema que me dé la geometría, las XSs y el resultado esperado. 300 páginas y ni una sola figura para mallar.

**Reply by marquezj** [reply]
Bueno, es una recopilación de soluciones analíticas o semi-analíticas de la ecuación de transporte. Si querés verificar que tu método numérico resuelve la ecuación de transporte (independientemente de otras soluciones numéricas), es contra lo que tenés que comparar. Lamentablemente no hay muchas soluciones de ese tipo.

Si querés comparar contra otros códigos, podés buscar ejemplos clásicos de benchmarks 2D como [C5G7-MOX](https://www.oecd-nea.org/science/wprs/eg3drtb/NEA-C5G7MOX.PDF), el benchmark de [VENUS-MOX](https://www.oecd-nea.org/jcms/pl_13442/benchmark-on-the-venus-2-mox-core-measurements?details=true) o los benchmarks numéricos que utilizó HyeongKae Park en su [tesis de doctorado](https://smartech.gatech.edu/handle/1853/13944).
